### ISSUE TITLE
parquet-251 causes the data read to be incorrect.

### ISSUE THREAD

**Issue by lyy-pineapple** [facebookincubator/velox#10747]
### Bug description

If the Parquet version is below 1.8.0, the statistics may be incorrect, leading to row group filtering issues and making the data unreadable.  ([parquet-251](https://issues.apache.org/jira/browse/PARQUET-251?jql=project%20%3D%20PARQUET%20AND%20resolution%20%3D%20Fixed%20ORDER%20BY%20created%20ASC%2C%20cf%5B12314422%5D%20ASC%2C%20cf%5B12310310%5D%20ASC%2C%20priority%20DESC%2C%20updated%20DESC))

### System information

Velox System Info v0.0.2
Commit: 3fbe31d01e714e553d7f7d16623980f8a78ebe8b
CMake Version: 3.25.1
System: Linux-5.4.0-150-generic
Arch: x86_64
C++ Compiler: /opt/rh/devtoolset-9/root/usr/bin/c++
C++ Compiler Version: 9.3.1
C Compiler: /opt/rh/devtoolset-9/root/usr/bin/cc
C Compiler Version: 9.3.1

**Comment by assignUser** [comment]
This is a parquet-java bug that was fixed ~10 years ago. Could you elaborate in what way this affects velox?

**Comment by lyy-pineapple** [comment]
If the version of Parquet used for writing is lower than 1.8.0, the row group statistics may be inaccurate, potentially leading to the filtering of row groups that fall within the data range. [Parquet-java](https://github.com/apache/parquet-java/blob/d4384d3f2e7703fab6363fd9cd80a001e9561db2/parquet-column/src/main/java/org/apache/parquet/CorruptStatistics.java#L57), when retrieving statistics, will determine whether to use them based on the version. Especially when we read data that was written by Hive or versions of Spark below 2.0

**Comment by assignUser** [comment]
Thanks. So it's basically about compatability with data that was written with these ancient (8-10 year old) versions? Depending on how much effort it is to maintain this fix I'd probably be against it. But I am also not to familar with how relevant this really is today etc. so I'll someone else deciede ^^

**Comment by lyy-pineapple** [comment]
When this issue occurs, there are no exceptions to notify the user, making it difficult to detect that incorrect data has been read. Due to historical reasons, some Hive or older version jobs are still in use, so we have implemented an internal fix.

**Comment by lyy-pineapple** [comment]
> When this issue occurs, there are no exceptions to notify the user, making it difficult to detect that incorrect data has been read. Due to historical reasons, some Hive or older version jobs are still in use, so we have implemented an internal fix.

Therefore, I recommend either fixing the issue as parquet-java did, or throwing an exception to indicate incompatibility and prevent data errors.

**Comment by assignUser** [comment]
>  there are no exceptions to notify the user, making it difficult to detect that incorrect data

Yeah that's bad. Either a fix or a clear error message would be good I agree. @kgpai
